main/src/library/HashSet.flix

```diff
@@ -0,0 +1,546 @@
+/*
+ * Copyright 2017 Liam Palmer, Esben Bjerre
+ * Copyright 2024 Jonathan Lindegaard Starup
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+///
+/// The HashSet type.
+///
+/// A set is currently represented internally as a red-black tree.
+///
+pub enum HashSet[t] with Sendable, Eq {
+    case HashSet(RedBlackTree[Int32, HashSet.ListSet[t]])
+}
+
+instance ToString[HashSet[t]] with ToString[t] {
+    pub def toString(x: HashSet[t]): String = HashSet.toString(x)
+}
+
+instance Collectable[HashSet[a]] with Eq[a], Hash[a] {
+    type Elm = a
+    pub def collect(iter: Iterator[a, ef, r]): HashSet[a] \ { ef, r } =
+        HashSet.collectI(iter)
+}
+
+instance Hash[HashSet[t]] with Hash[t] {
```

Comment:
Can we just derive this?

Reply:
oh maybe, didn't know that hash is derivable